buildengine/engine.go

```diff
@@ -278,10 +278,18 @@ func (e *Engine) watchForModuleChanges(ctx context.Context, period time.Duration
 	})
 
 	// Watch for file and schema changes
+	didUpdateDeployments := false
 	for {
+		var completedUpdatesTimer <-chan time.Time
+		if didUpdateDeployments {
```

Comment:
this if statement just allows us to not aggressively poll when in an idle state, rather than putting `time.After` directly in the select statement